### PR TITLE
Math overflow checking

### LINKED ISSUE #282: Implement math overflow checking
This should be an optional feature which can be enabled on the command line, with something like `solang --math-overflow`. 

The following code:

```
contract overflow {
        function foo(uint64 x) internal {
                uint64 y = x - 1;
        }

        function bar() public {
                foo(0);
        }
}
```

Calling `bar()` should fail with an assertion failure if `--math-overflow` has been specified. Overflow checking should be done for:

- Addition (both signed and unsigned)
- Subtraction (both signed and unsigned)
- Division/Multiplication/Modulo (signed and unsigned, up-to 64 bit types)
- Power does not need an implementation yet

Tests are required for all types.

This issue is purely for llvm overflow checking, see http://llvm.org/docs/LangRef.html#arithmetic-with-overflow-intrinsics

Multiplication, division, modulo larger than 64 bit and power have their own implementations which can be implemented in a separate issue.

### PR DESCRIPTION
**Pull request by KiChjang**
Fixes #282.

### CODE REVIEW

File: src/parser/solidity.rs [hunk 1]
```diff
@@ -1,4 +1,4 @@
-// auto-generated: "lalrpop 0.19.1"
+// auto-generated: "lalrpop 0.19.0"
```
Comment (seanyoung) [inline src/parser/solidity.rs:1]: Your parser was generated using an older version of lalrpop. Please remove this change (you can remove `Cargo.lock` and rebuild).

File: src/sema/contracts.rs [hunk 2]
```diff
@@ -15,11 +15,18 @@ use crate::emit;
 
 impl ast::Contract {
     /// Create a new contract, abstract contract, interface or library
-    pub fn new(name: &str, ty: pt::ContractTy, tags: Vec<ast::Tag>, loc: pt::Loc) -> Self {
+    pub fn new(
+        name: &str,
+        ty: pt::ContractTy,
+        tags: Vec<ast::Tag>,
+        loc: pt::Loc,
+        math_overflow_check: bool,
```
Comment (seanyoung) [inline src/sema/contracts.rs:23]: I don't think this new parameter is necessary. The optimization is also not specified, which is also a codegen issue (and not sema).

**Review by seanyoung — CHANGES_REQUESTED** [review]
Looks great, just two minor details.

Thank you very much!

File: src/parser/solidity.rs (hunk 1 above)
Comment (KiChjang) [reply src/parser/solidity.rs:1]: Haha, I ended up generating another file using a newer version of lalrpop. Is this ok?

Comment (seanyoung) [reply src/parser/solidity.rs:1]: of course

**Review by seanyoung — APPROVED** [review]

**Comment by seanyoung** [comment]
Very nice, thank you very much!

**Pull request merged**